src/test/java/com/thebois/models/world/structures/HouseTests.java

```diff
@@ -33,4 +39,106 @@ public void getTypeEqualsHouse() {
         assertThat(structure.getType()).isEqualTo(StructureType.HOUSE);
     }
 
+    @Test
+    public void deliverItemTest() {
+        // Arrange
+        final House house = new House(new Position());
+        final IItem item = new IItem() {
+        };
+
+        // Act
+
+        // Assert
+        assertThat(house.deliverItem(item)).isTrue();
+    }
+
+    @Test
+    public void builtStatusTest() {
+        // Arrange
+        final House house = new House(new Position());
+        final IItem item = new IItem() {
+        };
+
+        // Act
+        house.deliverItem(item);
+
+        // Assert
+        assertThat(house.builtStatus()).isEqualTo(0.01f);
+    }
+
+    @Test
+    public void neededItemsEqualTest() {
+        // Arrange
+        final House house = new House(new Position());
+        final IItem mockItem1 = Mockito.mock(IItem.class);
+        final IItem mockItem2 = Mockito.mock(IItem.class);
+        final IItem mockItem3 = Mockito.mock(IItem.class);
+
+        final Collection<IItem> neededItems = List.of(mockItem1, mockItem2, mockItem3);
+        final Collection<IItem> delivered = List.of(mockItem1);
+        final Collection<IItem> expectedNeededItems = List.of(mockItem2, mockItem3);
+
+        house.setAllNeededItems(neededItems);
+        house.setDeliveredItems(delivered);
+
+        // Act
+
+        // Assert
+        assertThat(house.neededItems()).isEqualTo(expectedNeededItems);
+    }
+
+    @Test
+    public void neededItemsNotEqualTest() {
+        // Arrange
+        final House house = new House(new Position());
+        final IItem mockItem1 = Mockito.mock(IItem.class);
+        final IItem mockItem2 = Mockito.mock(IItem.class);
+        final IItem mockItem3 = Mockito.mock(IItem.class);
```

Comment:
I mentioned in another test that `positionA` `positionB` `positionC` are bad names that make it hard to understand the test.

In this case, it is completely okay to call them `mockItem1` and so on. Why? Because I don't need to know which item is which to understand the test. It doesn't matter because I just need to know that one of the items is delivered and the other two should still be needed.

In the other test with the badly named positions, I needed to know which position was which in order to comprehend how the test worked, which made it really difficult to understand.